e2e/v3/jobs3/allocexec3.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package jobs3
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	nomadapi "github.com/hashicorp/nomad/api"
+	"github.com/shoenig/test/must"
+)
+
+// Exec executes cmd in the given task of a random allocation of the given
+// group.
+func (sub *Submission) Exec(group, task string, cmd []string) Logs {
+	queryOpts := sub.queryOptions()
+	jobsAPI := sub.nomadClient.Jobs()
+	stubs, _, err := jobsAPI.Allocations(sub.jobID, false, queryOpts)
+	must.NoError(sub.t, err, must.Sprintf("failed to query allocations for %s", group))
+
+	var allocID string
+	for _, stub := range stubs {
+		if stub.TaskGroup == group {
+			allocID = stub.ID
+			break
+		}
+	}
+	must.NotEq(sub.t, "", allocID, must.Sprintf("no allocation found for %s", group))
+
+	// do stuff
+	allocsAPI := sub.nomadClient.Allocations()
+	alloc, _, err := allocsAPI.Info(allocID, queryOpts)
+	must.NoError(sub.t, err, must.Sprintf("failed to query allocation %s", allocID))
```

Comment:
Is this bit just debugging or is the idea that we're asserting there's a real allocation? Wouldn't that just return an error below if there wasn't?

Reply:
I dunno, Rob Pike told me never to ignore a returned `err` so there we go

Reply:
I meant more why query it at all because I didn't think we used the `alloc`. But I missed that we're using it in the `Exec` call later. :facepalm: